vitrina/uapi/views/template_views.py

```diff
@@ -276,3 +286,42 @@ def delete(self, request: WSGIRequest, *args: Any, **kwargs: Any) -> HttpRespons
 
     def get_success_url(self) -> str:
         return reverse("agent-list", kwargs={"organization_id": self.organization.id})
+
+
+class RequestDetailView(BaseAgentView):
+    template_name = "requests/request_detail.html"
+    model = RequestHistory
+
+    def setup(self, request, *args, **kwargs) -> None:
+        self.object = get_object_or_404(
+            RequestHistory.objects.select_related("agent__organization"),
+            pk=kwargs.get("pk"),
+        )
+        kwargs[self.organization_url_kwarg] = self.object.agent.organization.pk
```

Comment:
```suggestion
        kwargs[self.organization_url_kwarg] = self.object.agent.organization_id
```
OPTIONAL
May save a join (select_related).